src/App.tsx

```diff
@@ -15,36 +16,101 @@
   'Garlic',
 ];
 
+enum SortBy {
+  None,
+  Abc,
+  Length,
+}
+
+enum Reverse {
+  No,
+  Yes,
+}
+
+const sortGoods = (sortField: SortBy, reverseField: Reverse) => {
+  const sortedGoods = [...goodsFromServer];
+
+  switch (sortField) {
+    case SortBy.Abc:
+      sortedGoods.sort((a, b) => a.localeCompare(b));
+      break;
+
+    case SortBy.Length:
+      sortedGoods.sort((a, b) => a.length - b.length);
+      break;
+
+    default:
+  }
+
+  if (reverseField) {
+    sortedGoods.reverse();
+  }
+
+  return sortedGoods;
+};
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = React.useState(SortBy.None);
+  const [reverseField, setReverseField] = React.useState(Reverse.No);
+  const goods: string[] = sortGoods(sortField, reverseField);
+
+  const reset = () => {
+    setSortField(SortBy.None);
+    setReverseField(Reverse.No);
+  };
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={cn('button is-info', {
+            'is-light': sortField !== SortBy.Abc,
+          })}
+          onClick={() => setSortField(SortBy.Abc)}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          type="button"
+          className={cn('button is-success', {
+            'is-light': sortField !== SortBy.Length,
+          })}
+          onClick={() => setSortField(SortBy.Length)}
+        >
           Sort by length
         </button>
 
-        <button type="button" className="button is-warning is-light">
+        <button
+          type="button"
+          className={cn('button is-warning', {
+            'is-light': reverseField !== Reverse.Yes,
+          })}
+          onClick={() =>
+            setReverseField(reverseField ? Reverse.No : Reverse.Yes)
```

Comment:
The logic for toggling `reverseField` is incorrect. The current logic will always set `reverseField` to `Reverse.No` because `reverseField` is an enum and any non-zero enum value is truthy. You should explicitly check for `Reverse.Yes` to toggle between `Reverse.No` and `Reverse.Yes`.